### PR TITLE
Added option to allow items in table to be selected

### PR DESCRIPTION
**Pull request by emilverwoerd**
Changed uif-table and added property uif-selected-items which allows to get information about the selected records in the table

### CODE REVIEW

File: src/components/table/tableDirective.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface ITableScope extends angular.IScope {
   rows: ITableRowScope[];
   tableType: string;
   tableTypeClass: string;
+  selectedItems: any;
```
Comment (andrewconnell) [inline src/components/table/tableDirective.ts:28]: Please add documentation in the JSDOC for this interface to document this property.

File: src/components/table/tableDirective.ts [hunk 2]
```diff
@@ -104,6 +101,7 @@ class TableController {
 export interface ITableAttributes extends angular.IAttributes {
   uifRowSelectMode?: string;
   uifTableType?: string;
+  uifSelectedItems?: any;
```
Comment (andrewconnell) [inline src/components/table/tableDirective.ts:108]: Please add documentation in the JSDOC for this interface to document this property.

**Review by andrewconnell — CHANGES_REQUESTED** [review]
Thanks for the new feature, but there are two things that need to be addressed with this PR before we can merge it in.

1. Update the JSDOC's for the interfaces that you added properties for.
1. Update the demo to reflect the new functionality that you added.

Without these two changes, this feature is not visible to consumers unless they poke around the source code and tests. Document the enhancement please.

**Comment by emilverwoerd** [comment]
Going to implement the requested changes. Also the examples are faulty with the controller on a `<p>` tag will fix that also.

**Review by andrewconnell — APPROVED** [review]

**Comment by emilverwoerd** [comment]
Hi Andrew thanks for adding but is there a reason why the .spec file isn't updated

**Comment by andrewconnell** [comment]
It isn't? I merged your changes... what's missing? If there's an issue, can you create an issue, self-assign and a new PR that closes it?

I had to squash your commits into one as we only merge a PR as a single commit... I might have made a mistake in that as there were a lot of conflicts between your multiple commits.

**Comment by emilverwoerd** [comment]
well the file 'tableDirective.spec.ts' wasn't updated for the rest it looks quite good so I could add a PR and fix it.

**Comment by andrewconnell** [comment]
My bad... would you mind creating an issue documenting the problem? While I let this one pass, we really want every PR to be tied to an issue.

**Pull request closed without merging**